presto-main/src/main/java/com/facebook/presto/security/AccessControlManager.java

```diff
@@ -150,6 +150,15 @@ public void checkCanSetUser(Optional<Principal> principal, String userName)
         authenticationCheck(() -> systemAccessControl.get().checkCanSetUser(principal, userName));
     }
 
+    @Override
+    public void checkQueryIntegrity(Identity identity, String query)
+    {
+        requireNonNull(identity, "extraCredentials is null");
```

Comment:
While I am not completely sure about the context behind the comment that you linked to, I feel that it is reasonable to validate that the input that the class is getting from an external agent. So I can generalize it to when something out of the class creates an instance by calling the constructor or to validate arguments passed to a public method of a class.

Reply:
@mayankgarg1990 , definitely I see the need for null checking params in public methods.
However seems like in presto we usually do not null check method parameters using `requireNonNull`. We do that only for constructor arguments. cc @mbasmanova .

Reply:
@ajaygeorge - If you take a look `AccessControlManager` that has been doing this since 2015 - so there are examples already in the codebase. Also, it is better that we focus on what makes sense rather than how things are being done.